Add an environment variable to python-sip that extends its search
directories for .sip files.

It seems that we cannot easily change the destination folder of these
files though, so this variable must be set on a per-package basis (and
non through search-path).

--- sip/sipbuild/builder.py
+++ sip/sipbuild/builder.py
@@ -227,5 +227,9 @@
                     os.path.join(project.target_dir,
                             project.get_bindings_dir()))
 
+        # Add extra bindings from environment for GNU Guix
+        if 'SIP_INCLUDE_DIRS' in os.environ:
+            sip_include_dirs.extend(os.environ['SIP_INCLUDE_DIRS'].split(os.pathsep))
+
         # Generate the code for each set of bindings.
         api_files = []
